Lib/urllib/robotparser.py

```diff
@@ -186,15 +186,17 @@ def crawl_delay(self, useragent):
         for entry in self.entries:
             if entry.applies_to(useragent):
                 return entry.delay
-        return self.default_entry.delay
+        if self.default_entry:
+            return self.default_entry.delay
 
```

Comment:
Both `crawl_delay` and `request_rate` should return `None` explicitly if there is no applicable entry.